subsys/net/openthread/rpc/common/ot_rpc_common.c

```diff
@@ -39,6 +37,112 @@ void ot_rpc_decode_dataset_tlvs(const struct nrf_rpc_group *group, struct nrf_rp
 	}
 }
 
+static void ot_rpc_encode_timestamp(struct nrf_rpc_cbor_ctx *ctx, const otTimestamp *timestamp)
+{
+	nrf_rpc_encode_uint64(ctx, timestamp->mSeconds);
+	nrf_rpc_encode_uint(ctx, timestamp->mTicks);
+	nrf_rpc_encode_bool(ctx, timestamp->mAuthoritative);
+}
+
+static void ot_rpc_decode_timestamp(struct nrf_rpc_cbor_ctx *ctx, otTimestamp *timestamp)
+{
+	timestamp->mSeconds = nrf_rpc_decode_uint64(ctx);
+	timestamp->mTicks = nrf_rpc_decode_uint(ctx);
+	timestamp->mAuthoritative = nrf_rpc_decode_bool(ctx);
+}
+
+void ot_rpc_encode_dataset(struct nrf_rpc_cbor_ctx *ctx, const otOperationalDataset *dataset)
+{
+	ot_rpc_encode_timestamp(ctx, &dataset->mActiveTimestamp);
+	ot_rpc_encode_timestamp(ctx, &dataset->mPendingTimestamp);
+	nrf_rpc_encode_buffer(ctx, dataset->mNetworkKey.m8, sizeof(dataset->mNetworkKey.m8));
+	nrf_rpc_encode_str(ctx, dataset->mNetworkName.m8, strlen(dataset->mNetworkName.m8));
+	nrf_rpc_encode_buffer(ctx, dataset->mExtendedPanId.m8, sizeof(dataset->mExtendedPanId.m8));
+	nrf_rpc_encode_buffer(ctx, dataset->mMeshLocalPrefix.m8,
+			      sizeof(dataset->mMeshLocalPrefix.m8));
+	nrf_rpc_encode_uint(ctx, dataset->mDelay);
+	nrf_rpc_encode_uint(ctx, dataset->mPanId);
+	nrf_rpc_encode_uint(ctx, dataset->mChannel);
+	nrf_rpc_encode_buffer(ctx, dataset->mPskc.m8, sizeof(dataset->mPskc.m8));
+
+	nrf_rpc_encode_uint(ctx, dataset->mSecurityPolicy.mRotationTime);
```

Comment:
nit: another approach could be to encode the bitfields in `uint16_t` using well-defined masks. A bit dirtier code but lower CBOR overhead. But your approach is OK as well.